app/src/androidTest/java/org/mozilla/fenix/ui/BadServerResponseTest.kt

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.ui
+
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.After
+import org.junit.Test
+import org.mozilla.fenix.helpers.HomeActivityTestRule
+import org.mozilla.fenix.helpers.MockWebServerHelper
+import org.mozilla.fenix.ui.robots.homeScreen
+
+class BadServerResponseTest {
```

Comment:
nit: this class could use kdoc to explain why we want to test this test case. Perhaps we should link to the bug for the crash where this was actually a problem https://github.com/mozilla-mobile/fenix/issues/17762.